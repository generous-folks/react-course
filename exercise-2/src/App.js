import React, { useState, useEffect } from 'react';
import { getArticles } from './utils/api.utils';

export default function App() {
  const [articles, setArticles] = useState([]);

  useEffect(() => {
    try {
      if (articles.length === 0) {
        getArticles()
          .then(res => setArticles(res))
          .catch(err => console.error(err));
      }
    } catch (error) {
      throw new Error(`Fetch articles failed: ${error.message}`);
    }
  }, [articles]);

  return (
    <div>
      <h2>Home Page</h2>
      <div>
        <h4>Articles</h4>
        <ul>{articles.length > 0 && articles.map(({ id, name }) => <li key={id}>{name}</li>)}</ul>
      </div>
    </div>
  );
}
