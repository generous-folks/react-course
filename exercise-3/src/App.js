import React from 'react';
import { Switch, Route } from 'react-router-dom';

import { HomePage } from './pages/home.page';
import { AboutPage } from './pages/about.page';

export default function App() {
  return (
    <Switch>
      <Route path="/about">
        <AboutPage />
      </Route>
      <Route exact path="/">
        <HomePage />
      </Route>
    </Switch>
  );
}
