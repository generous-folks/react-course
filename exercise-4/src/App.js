import React from 'react';
import { Switch, Route } from 'react-router-dom';

import { HomePage } from './pages/home.page';
import { AboutPage } from './pages/about.page';
import { ContactPage } from './pages/contact.page';

export default function App() {
  return (
    <Switch>
      <Route path="/about">
        <AboutPage />
      </Route>
      <Route path="/contact">
        <ContactPage />
      </Route>
      <Route exact path="/">
        <HomePage />
      </Route>
    </Switch>
  );
}
