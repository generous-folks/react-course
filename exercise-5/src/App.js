import React from 'react';
import { Switch, Route, BrowserRouter as Router } from 'react-router-dom';

import { HomePage } from './pages/home.page';
import { AboutPage } from './pages/about.page';
import { ContactPage } from './pages/contact.page';

export default function App() {
  return (
    <Router>
      <Switch>
        <Route path="/about">
          <AboutPage />
        </Route>
        <Route path="/contact">
          <ContactPage />
        </Route>
        <Route exact path="/">
          <HomePage />
        </Route>
      </Switch>
    </Router>
  );
}
