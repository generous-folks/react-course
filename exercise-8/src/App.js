import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { UserProvider } from './modules/user/user.context';

import { AppRoutes } from './modules/routing/components/routes.component';

export default function App() {
  return (
    <UserProvider>
      <Router>
        <AppRoutes />
      </Router>
    </UserProvider>
  );
}
