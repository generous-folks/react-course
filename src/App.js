import React from "react";
import { Switch, Route } from "react-router-dom";

import { Home } from "./pages/home.page";

export default function App() {
  return (
        <Switch>
          <Route path="/about">
            {/* <About /> */}
          </Route>
          <Route path="/articles/:id">

          </Route>
          <Route path="/">
            <Home />
          </Route>
        </Switch>
  );
}
