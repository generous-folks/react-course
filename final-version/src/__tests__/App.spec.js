import React from 'react';
// import { act } from 'react-dom/test-utils';
import { shallow } from 'enzyme';

import App from '../App';

// jest.mock('react', () => global.mockReactWithHooks({ effect: true, state: true }));

describe('App', () => {
  // it('renders user data', async () => {
  //   const fakeUser = {
  //     name: 'Joni Baez',
  //     age: '32',
  //     address: '123, Charming Avenue',
  //   };

  //   jest.spyOn(global, 'fetch').mockImplementation(() =>
  //     Promise.resolve({
  //       json: () => Promise.resolve(fakeUser),
  //     }),
  //   );

  //   // Use the asynchronous version of act to apply resolved promises
  //   await act(async () => {
  //     render(<User id="123" />, container);
  //   });

  //   expect(container.querySelector('summary').textContent).toBe(fakeUser.name);
  //   expect(container.querySelector('strong').textContent).toBe(fakeUser.age);
  //   expect(container.textContent).toContain(fakeUser.address);

  //   // remove the mock to ensure tests are completely isolated
  //   global.fetch.mockRestore();
  // });

  it('should render correctly', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
