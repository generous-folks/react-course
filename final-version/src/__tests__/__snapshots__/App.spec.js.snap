// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      path="/about"
    >
      <AboutPage />
    </Route>
    <Route
      path="/contact"
    >
      <ContactPage />
    </Route>
  </Switch>
  <ArticlesProvider>
    <CartProvider>
      <Switch>
        <Route
          path="/articles/:id"
        >
          <ArticlePage />
        </Route>
        <Route
          exact={true}
          path="/"
        >
          <HomePage />
        </Route>
      </Switch>
    </CartProvider>
  </ArticlesProvider>
</BrowserRouter>
`;
